### ISSUE TITLE
error not displayed when connection to the db fails

### ISSUE THREAD

**Issue by richlv** [whosgonna/Zabbix-Tiny.pm#12]
zabbix 3.0 frontend fails with :
"Database error
    Error connecting to database: No such file or directory"

api in that case returns :

```
{"jsonrpc":"2.0","error":{"code":1,"message":"Error` connecting to database: No such file or directory","data":""},"id":1}
```

zabbix::tiny use was like this :

```
my $zabbix = Zabbix::Tiny->new(
server   => $url,
```

that fails with `Error:  at stuff.pl line 33.`

**Comment by whosgonna** [comment]
Note to self, "stuff.pl" is the name of Richlv's script, not part of my error message.

Richlv,  can you provide any more instruction on how to reproduce? I tried this:

```
#!/usr/bin/env perl
use strict;
use warnings;
use lib "../lib/";
use Zabbix::Tiny;

# Create a new Zabbix::Tiny object
my $zabbix = Zabbix::Tiny->new(
    server   => 'https://zabbix.local/api_jsonrpc.php',
);
```

The failure I received was:  `Missing required arguments: password, user at (eval 16) line 51.`

**Comment by whosgonna** [comment]
Note, that the line reflected there is from the module, so I can look at cleaning it up to show the calling line, but as a simple explanation this is clear, so I must be doing something slightly different.

**Comment by richlv** [comment]
oh, sorry for not being clear enough - the script is a proper one, it also passed user & parameter, then tried to execute a method against the api.
the main idea is that any api-using script should show the same problem when the api returns an error like that.

**Comment by whosgonna** [comment]
I agree, but did you try an invalid method, have invalid parameters, etc?  How can I reproduce this failure?  Is it limited to 3.0?  I don't have much access to a 3.0 server right now, but if there's a specific enough scenario to reproduce, I should be able to do that.

**Comment by richlv** [comment]
the backend database was down ;)

**Comment by whosgonna** [comment]
Ahhh  - well, then the fact that the only 3.0 server I have is in a relatively unused scenario, I'm actually in a GOOD place to test this :)

**Comment by whosgonna** [comment]
Fixed in [027e78a](https://github.com/whosgonna/Zabbix-Tiny.pm/commit/027e78a9299bab3e0d0f52ee35f749537e73422d)
